aquarium-optimized/src/dawn/FishModelDawn.cpp

```diff
@@ -174,10 +199,22 @@ void FishModelDawn::draw()
     pass.SetVertexBuffers(4, 1, &binormalBuffer->getBuffer(), vertexBufferOffsets);
     pass.SetIndexBuffer(indicesBuffer->getBuffer(), 0);
 
-    for (int i = 0; i < instance; i++)
+    if (enableDynamicBufferOffset)
+    {
+        for (int i = 0; i < instance; i++)
+        {
+            uint64_t offset = 256u * i;
+            pass.SetBindGroup(3, bindGroupPers[0], 1, &offset);
```

Comment:
For Vulkan now, This may hit the limit of std::numeric_limit<uint32_t>::max() if the fish number large enough.

Reply:
More details here : https://github.com/KhronosGroup/Vulkan-Docs/issues/971

Reply:
Checked that it won't  hit the limit if rendering 100000 fishes.